src/puppetlabs/puppetdb/http/command.clj

```diff
@@ -21,21 +21,31 @@
 
 (def valid-commands-str (str/join ", " (sort (vals command-names))))
 
+(defn coerce-to-int
+  [s]
+  (if (number? s) s (Integer. s)))
+
 (defn validate-command-version
   [app]
-  (fn [{:keys [body-string] :as req}]
-    (let [{:keys [command version]} (json/parse-string body-string true)
+  (fn [{:keys [body-string params param-post?] :as req}]
+    (let [{:strs [command version]} (if param-post?
+                                      params
+                                      (json/parse-string body-string))
+          command (str/replace command "-" " ")
```

Comment:
Ticket made PDB-2311, but a note from the channel... Should these be underscores instead of dashes?